core/driver/sources/src/test/scala/org/virtuslab/ideprobe/dependencies/ErrorValidatorTest.scala

```diff
@@ -0,0 +1,119 @@
+
+  private val errorIdeMessages: List[IdeMessage] = List(messageBusConnectionImplNPE, uastMetaLanguageNPE)
+    .map(content => IdeMessage(level = Level.Error, content, None))
+
+  private val messageBusSpecificString = "at com.intellij.util.messages.impl.MessageBusConnectionImpl"
+  private val uastMetaSpecificString = "at com.intellij.uast.UastMetaLanguage"
+
+  @Test
+  def shouldIncludeAllErrorsByDefault(): Unit = {
+    val exceptions: List[Option[Exception]] = collectExceptions(defaultCheckConfigWithErrorsEnabled)
+    val exceptionsIncluded = exceptions.filter(_.nonEmpty)
+    val exceptionsIgnored = exceptions.filter(_.isEmpty)
+    assertEquals(2, exceptionsIncluded.size)
+    assertEquals(0, exceptionsIgnored.size)
+  }
+
+  @Test
+  def shouldIgnoreAllErrorsIfNotEnabled(): Unit = {
+    val checkConfigErrorsDisabled = defaultCheckConfigWithErrorsEnabled.copy(errors =
+      ErrorConfig(
+        enabled = false,
+        includeMessages = defaultCheckConfigWithErrorsEnabled.errors.includeMessages,
+        excludeMessages = defaultCheckConfigWithErrorsEnabled.errors.excludeMessages
+      )
+    )
+    val exceptions: List[Option[Exception]] = collectExceptions(checkConfigErrorsDisabled)
+    val exceptionsIncluded = exceptions.filter(_.nonEmpty)
+    val exceptionsIgnored = exceptions.filter(_.isEmpty)
+    assertEquals(0, exceptionsIncluded.size)
+    assertEquals(2, exceptionsIgnored.size)
+  }
+
+  @Test
+  def shouldIncludeOnlySpecificErrorsIfIncludeMessagesConfigured(): Unit = {
+    val checkConfigForMessageBusErrorsOnly = defaultCheckConfigWithErrorsEnabled.copy(errors =
+      ErrorConfig(
+        enabled = defaultCheckConfigWithErrorsEnabled.errors.enabled,
+        includeMessages = Seq(s".*$messageBusSpecificString.*"),
+        excludeMessages = defaultCheckConfigWithErrorsEnabled.errors.excludeMessages
+      )
+    )
+    val exceptions: List[Option[Exception]] = collectExceptions(checkConfigForMessageBusErrorsOnly)
+    val exceptionsIncluded = exceptions.filter(_.nonEmpty)
+    val exceptionsIgnored = exceptions.filter(_.isEmpty)
+    assertEquals(1, exceptionsIncluded.size)
+    assertEquals(1, exceptionsIgnored.size)
+    assertTrue(exceptionsIncluded.head.get.getMessage.contains(messageBusSpecificString))
+  }
+
+  @Test
+  def shouldExcludeSpecificErrorsByRegexUsage(): Unit = {
+    val checkConfigToIgnoreMessageBusErrors = defaultCheckConfigWithErrorsEnabled.copy(errors =
+      ErrorConfig(
+        enabled = defaultCheckConfigWithErrorsEnabled.errors.enabled,
+        includeMessages = defaultCheckConfigWithErrorsEnabled.errors.includeMessages,
+        excludeMessages = Seq(s".*$messageBusSpecificString.*")
+      )
+    )
+    val exceptions: List[Option[Exception]] = collectExceptions(checkConfigToIgnoreMessageBusErrors)
+    val exceptionsIncluded = exceptions.filter(_.nonEmpty)
+    val exceptionsIgnored = exceptions.filter(_.isEmpty)
+    assertEquals(1, exceptionsIncluded.size)
```

Comment:
I guess assertion logic in these tests could be less complicated. Just return `List[Exception]` and check if interesting ones are there and if count is right. Now each test is validating that if you pass a list of 2 elements to a helper method that you added in this test, it returns 2 results. Ofc it will, it just calls map and it is not a part of tested code :p

Reply:
> Now each test is validating that if you pass a list of 2 elements to a helper method that you added in this test, it returns 2 results. Ofc it will, it just calls map and it is not a part of tested code :p

Hm, I don't think your're right here. In each test `exceptionsIncluded.size` and `exceptionsIgnored .size` are checked. To make sure, that including / ignoring errors works fine. For example:
- `shouldIncludeAllErrorsByDefault()` - tests as follows:
```
assertEquals(2, exceptionsIncluded.size)
assertEquals(0, exceptionsIgnored.size)
```
- `shouldExcludeSpecificErrorsByRegexUsage()` - tests as follow:
```
assertEquals(1, exceptionsIncluded.size)
assertEquals(1, exceptionsIgnored.size)
```
and it is a part of tested code - we make sure that ErrorValidator includes / excludes errors properly with the config.

Or maybe I just didn't understand your comment properly?

Reply:
I mean, it is enough to check how many errors were found. Otherwise you are testing the test code. It is clear that `ignored count == 2 - error count` if you pass w errors. You don't need to complicate test code with optionals and get calls.

Reply:
ok, done as suggested